src/cmt_decode_prometheus.c

```diff
@@ -456,6 +456,9 @@ static int add_metric_histogram(struct cmt_decode_prometheus_context *context)
      * - sum
      * - count */
     bucket_count = cfl_list_size(&context->metric.samples) - 3;
+    if ((int)bucket_count < 0) {
+        abort();
```

Comment:
there is no circumstances where cmetrics will abort, being a library must handle/propagate the error, implement a default workaround or any other that does not make the caller program to exit.

Reply:
TBH I've added this just as a debug assertion, this should never happen (it did in the bug that caused fluent-bit 6534). If you want I can remove it, but the purpose of this is signal calling code that there's a bug in the library